src/pixmap.c

```diff
@@ -13,6 +13,10 @@ twin_pixmap_t *twin_pixmap_create(twin_format_t format,
                                   twin_coord_t height)
 {
     twin_coord_t stride = twin_bytes_per_pixel(format) * width;
+    /* Padding the stride to a multiple of 4 bytes */
```

Comment:
Why do you think it is necessary? Explain and consider to submit another pull request for generic pixmap manipulation.

Reply:
<img src="https://github.com/user-attachments/assets/a95737ff-1b51-4acb-95cf-0bba653d53be" width="400">

If we comment it, it will happen _pixman_log_error as follow:
```
*** BUG ***
*** BUG ***
In create_bits_image_internal: The expression bits == NULL || (rowstride_bytes % sizeof (uint32_t)) == 0 was false
Set a breakpoint on '_pixman_log_error' to debug

[1]    62492 segmentation fault (core dumped)  ./demo-sdl
```

pixman_image_t *msk fails to be allocated in [/src/pixman.c#L102](https://github.com/sysprog21/mado/commit/7ccf36edd9a40450add80dc154c345d2d66d2dfd#diff-5f7bbe04dc9ae37abc7b25930f4f97c723459d3c7791e04ec99804d4baf98e6fR102). Finally, pixman_image_unref will access a nullptr and happen segmentation fault in [/src/pixman.c#L107](https://github.com/sysprog21/mado/commit/7ccf36edd9a40450add80dc154c345d2d66d2dfd#diff-5f7bbe04dc9ae37abc7b25930f4f97c723459d3c7791e04ec99804d4baf98e6fR107)

Reply:
Do not use screenshots for plain text content, as this is inaccessible to visually impaired users.

Reply:
> If we comment it, it will happen _pixman_log_error as follow:
> ...
> `pixman_image_t *msk` fails to be allocated. Finally, `pixman_image_unref` will access a nullptr, and segmentation fault occurs.

Next, you should check the pointer alignment requirements for Pixman and consider providing the corresponding macros for alignment. See [mimalloc](https://github.com/microsoft/mimalloc/blob/dev/include/mimalloc/internal.h#L308-L324) for reference.